### PR TITLE
feat(FN-1719): add E2E test for removing fees from payment

### PR DESCRIPTION
**Pull request by sw-mattw**
## Introduction :pencil2:
This PR introduces end-to-end tests to validate the changes implemented in #3303 and #3295, specifically focusing on the journey to remove fees from a payment.

Minimal to test the core journeys (fee removed, error message displayed, status updated). More granular test cases in the unit tests.

## Resolution :heavy_check_mark:
- Add E2E tests to test the journey to remove fees from a payment.

## Miscellaneous :heavy_plus_sign:
N/A

### CODE REVIEW

File: e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/remove-fees-from-payment.spec.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+      cy.get(`[type="checkbox"][id="feeRecordId-${id}"]`).check();
+    });
+    pages.utilisationReportEditPaymentPage.clickRemoveSelectedPaymentsButton();
+    pages.utilisationReportEditPaymentPage.errorSummary().contains('You cannot remove all the fees. Delete the payment instead.');
+    feeRecordIds.forEach((id) => {
+      cy.get(`[type="checkbox"][id="feeRecordId-${id}"]`).should('be.checked');
+    });
+  });
+
+  it('should remove the selected fees from the payment after clicking the remove selected fees button', () => {
```
Comment (JoshBinns2000) [inline e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/remove-fees-from-payment.spec.js:84]: Could we possibly check that the other fee records still exist after we have removed this specific fee?

File: e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/remove-fees-from-payment.spec.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+    cy.get(`[type="checkbox"][id="feeRecordId-${feeRecordId}"]`).check();
+    pages.utilisationReportEditPaymentPage.clickRemoveSelectedPaymentsButton();
+
+    cy.url().should('eq', relative(`/utilisation-reports/${reportId}/edit-payment/${paymentId}`));
+    cy.get(`[type="checkbox"][id="feeRecordId-${feeRecordId}"]`).should('not.exist');
+  });
+
+  it(`should update the fee record status if another fee has been removed from the payment group`, () => {
+    const feeRecordId = feeRecordIds[0];
```
Comment (JoshBinns2000) [inline e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/remove-fees-from-payment.spec.js:95]: It looks like this is only being used in the checkbox selection below, but isn't actually being used to create either of the fee records below. I also don't think this test needs to rely on `feeRecordIds` and you can probably just hardcode `feeRecordId` to be some set value

**Review by JoshBinns2000 — COMMENTED** [review]
Nice tests - just a two small comments

File: e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/remove-fees-from-payment.spec.js (hunk 1 above)
Comment (sw-mattw) [reply e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/remove-fees-from-payment.spec.js:86]: Good point - made the change in 1018f9926faf81d4ac74ff707971a1e384418564 thanks

File: e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/remove-fees-from-payment.spec.js (hunk 2 above)
Comment (sw-mattw) [reply e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/remove-fees-from-payment.spec.js]: Yep use of feeRecordIds is unnecessary - resolved in 46ba4e10dd173f6bcd9d4388c168db454fa6191e thank you

File: e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/remove-fees-from-payment.spec.js [hunk 3]
```diff
@@ -71,29 +73,33 @@ context(`${PDC_TEAMS.PDC_RECONCILE} users can remove fees from payments`, () =>
...
+    feeRecordIds.forEach((feeRecordId) => {
+      getFeeRecordCheckbox(feeRecordId).should('be.checked');
     });
   });
 
   it('should remove the selected fees from the payment after clicking the remove selected fees button', () => {
-    const feeRecordId = feeRecordIds[0];
+    const getFeeRecordRow = (feeRecordId) => cy.get(`tbody tr[data-cy="fee-record-details-table-row--feeRecordId-${feeRecordId}"]`);
+    const feeRecordIdToRemove = feeRecordIds[0];
+    const otherFeeRecordIds = feeRecordIds.slice(1);
```
Comment (JoshBinns2000) [inline e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/remove-fees-from-payment.spec.js:89]: Nit but you can write this as `const [feeRecordIdToRemove, ...otherFeeRecordIds] = feeRecordIds`

**Review by JoshBinns2000 — DISMISSED** [review]
Approved - just left a comment around splitting the `feeRecordIds` array

File: e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/remove-fees-from-payment.spec.js (hunk 3 above)
Comment (sw-mattw) [reply e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/remove-fees-from-payment.spec.js]: Nice, missed that thanks for spotting - just pushed as fa4a32d841055b5a38a13bc8328c46a83b05cf36

**Review by JoshBinns2000 — APPROVED** [review]

**Pull request merged**